onnx/test/shape_inference_test.py

```diff
@@ -59,7 +60,9 @@ def _assert_inferred(self, graph, vis, **kwargs):  # type: (GraphProto, List[Val
         inferred_vis = list(inferred_model.graph.value_info)
         vis = list(sorted(vis, key=lambda x: x.name))
         inferred_vis = list(sorted(inferred_vis, key=lambda x: x.name))
-        if vis == inferred_vis:
+        # remove generated symbolic shape for numerical matching
+        removed_symbolic_inferred_vis = re.sub('\n.*dim_param: \"unk_.*', '', str(inferred_vis))
```

Comment:
What is this for? I think an alternative way of checking that doesn't assume what the generated names look like might be better.

Reply:
Initially it removed "dim_param: unk_XXX" for matching the inferred valueinfo and expected valueinfo. Now it will check each dim instead of matching the whole string. If it encounters symbolic shape, it will only check whether a dim_param is produced.